qingke-rt/src/lib.rs

```diff
@@ -120,12 +120,24 @@ unsafe extern "C" fn qingke_setup_interrupts() {
     // 0x3 both nested interrupts and hardware stack
     // 0x1 only hardware stack
 
+    // for user mode: mstatus = 0x80
+    // mpp(m-mode previous privilege) = 0b00 = U
+    // mpie(m-mode previous interrupt enable) = 0b1
+    // mie(m-mode interrupt enable) = 0b0
+    // interrupts will be enabled when mret at the end of handle_reset
+    // jumps to main (mret does mie = mpie)
+    // for machine mode: mstatus = 0x1880
+    // mpp = 0b11
+    // mpie = 0b1
+    // mie = 0b0
+
     // Qingke V2A, V2C
+    // (does not have user mode)
```

Comment:
non-blocking: `build.rs` can be used to check illegal feature combinations. I don't know if static_assert is avaliable.